lib/validators/specValidator.js

```diff
@@ -603,22 +596,35 @@ class SpecValidator {
       }
       let location = parameter.in;
       if (location === 'path' || location === 'query') {
+        if (location === 'path' && parameterValue && typeof parameterValue.valueOf() === "string") {
+          if (options.pathTemplate.charAt(options.pathTemplate.indexOf(`${parameter.name}`) - 2) === '/' && parameterValue.startsWith('/')) {
+            let msg = `In operation "${operation.operationId}", example for parameter "${parameter.name}": "${parameterValue}" starts with a forward slash ` +
+              `and the path template: "${options.pathTemplate}" contains a forward slash before the parameter starts. This will cause double forward slashes ` +
+              ` in the request url. Thus making it incorrect. Please rectify the example.`;
+            let e = self.constructErrorObject(ErrorCodes.DoubleForwardSlashesInUrl, msg);
+            result.validationResult.errors.push(e);
+            foundIssues = true;
+            break;
+          }
+          //replacing forward slashes with empty string because this messes up Sways regex validation of path segment.
+          parameterValue = parameterValue.replace(/\//ig, '');
```

Comment:
I found https://github.com/OAI/OpenAPI-Specification/issues/892. It appears that having path parameters with slashes may not be allowed in swagger??

Reply:
Yes they are technically not allowed. But in real world services have such urls and we need to support them.
Think about apis like dropbox or file storage apis where they have a file path or container path as a path parameter. They wont be able to use swagger.